meetings/2017-02-01.md

```diff
@@ -0,0 +1,261 @@
+fulfilled. But this isn’t how Babel does it.
+
+@bmeck looking for specific direction from CTC, asked for vote.
+
+@jasnell: should be a clear separation between require/CJS and import/ESM,
+perhaps support async for import but not require.
+
+@mylesborins: modules written for web will expect an async loader, and would be
+best for Node.js to match browser behavior as much as possible.
+
+@bmeck: browsers will move more and more to async loading.
+
+@addaleax: having an async loader matches expected behavior in other places,
+like top-level `await`.
+
+@bmeck: synchronous loader and top-level `await` are mutually exclusive, can’t
+do both.
+
+@ofrobots: if we go async in Node we will disturb ecosystem which expects all to
+be sync. Yet if we go sync we will lose compat with browsers and web.
+
+@bmeck: browsers don’t have the option to ship an async loader.
+
+@addaleax: an async loader wouldn’t effect CJS modules so concerns can be
+addressed as authors move to ES modules. So wouldn’t be so disruptive.
+
+@Fishrock123: Getting opinions from trainers of new users if this will make
+learning curve steeper. Would be best if `require` is only sync. Async loader
+could handle circular deps better, so that’s good.
+
+Suggest using require for CJS and import for ES.
+
+@Trott: We're voting on an EP not landing an implementation, which means that we
+can change course later if necessary.
+
+Are we debating changing `require` or just `import`?
+
+`import(‘es’)` will always return a Promise, `require(‘cjs’)` will always return
+an object. Question is `require(‘es’)` and `import(‘cjs’)`.
+
+@jasnell: supporting both async and sync will be difficult.
+
+@evanlucas: biggest concern is breaking ecosystem in order to support future
+browsers.
+
+@bmeck: but we’re breaking them anyway cause we can’t support named imports
+(which break Babel transpilation of ES).
+
+How will a sync loader affect future module usage and development in JS/TC39?
+* Would disallow top-level await.
+* Circular dependencies with asynchronously loaded ES modules.
+
+Q: Why does Node.js want to have synchronous imports?
+A: So that you can require an ES module and have it behave in the same way as a
+CJS module.
+
+@jasnell called for a vote next week, @Fishrock123 seconded.
+
+**Next steps**
+
+* Vote by next week.
+
+---
+
+### assert: enforce type check in deepStrictEqual [node#10282](https://github.com/nodejs/node/pull/10282)
+
+@Trott: Suggest allowing internal changes as long as they don’t change API
+surface.
+
+@Fishrock123: Current policy is not to change unless needed by core.
+
+Should we unlock `assert` for API changes?
+
+Perhaps not, but we should still support new ES constructs within existing API.
+
+@mylesborins will open an issue to re-evaluate what “locked” policy means,
+becuase JS language changes happen more frequently now.
+
+**Next steps** 
+
+* @Trott to draft proposal based on this discussion as PR.
```

Comment:
My ideas of the absolute minimum of what we ought to do here are in https://github.com/nodejs/node/pull/11113 (and perhaps also https://github.com/nodejs/node/pull/11112 although that's not really a content change). While I wouldn't necessarily object to bigger changes, this is what I consider more-or-less the minimum we can do to signal that it's OK to use `assert` as long as you don't need features that aren't there and that we are interested in fixing bugs in `assert` even if they don't affect Node.js core tests. Fixing bugs: yes. New features: no, but there's lots of great options in npm.